File: Code-Java/src/codejava/base/unordered_map/UnorderedMap.java
package codejava.base.unordered_map;

import codejava.base.ExampleObject;

import java.util.HashMap;

public class Unordered_map {
    private static HashMap<Integer, ExampleObject> hashmap_;
    private static ExampleObject obj;

    public static void UnorderedMapProgram()
    {
        hashmap_ = new HashMap<Integer, ExampleObject>();
        obj = new ExampleObject();
        boolean isEmpty = hashmap_.isEmpty();
        int size = hashmap_.size();
        hashmap_.clear();
        hashmap_.put(1,obj);
        boolean isContained = hashmap_.containsKey(1);
        ExampleObject findObj = hashmap_.get(1);
        hashmap_.remove(1);
    }
}

File: Code-Cpp/code_cpp/base/unordered_map/UnorderedMap.cpp
#include "Unordered_map.h"

std::unordered_map<int,ExampleObject> Unordered_map::hashmap_;
ExampleObject Unordered_map::obj;

void Unordered_map::UnorderedMapProgram() {
    hashmap_ = std::unordered_map<int,ExampleObject>();
    obj = ExampleObject();
    bool isEmpty = hashmap_.empty();
    int size = hashmap_.size();
    hashmap_.clear();
    hashmap_.insert({1,obj});
    bool isContained = hashmap_.contains(1);
    ExampleObject findObj = hashmap_[1];
    hashmap_.erase(1);
}
